### PR TITLE
[CORE-1042] Cache Symlink Commit Uploads

### PR DESCRIPTION
**Pull request by FahadBSyed**
## Description
This PR resolves a bug reported in CORE-1042 by expanding the cacheClient to wrap the ModifyFile client's copyFile(). Instead of copying files in a commit, a commit ID is put into a cache and a fileSet is generated for each commit. This saves time by bypassing calls to Postgres and Auth.

### CODE REVIEW

**Comment by acohen4** [comment]
What do we think the main pathological cases here were? Is it the GetFileSet RPC (network + DB I/O)?

**Comment by FahadBSyed** [comment]
> What do we think the main pathological cases here were? Is it the GetFileSet RPC (network + DB I/O)?

I think its the overhead of auth and postgres calls per copy() operation whereas with creating a fileSet once, we can just pull that from the cache over and over again.

File: src/internal/pfssync/cache_client.go [hunk 1]
```diff
@@ -66,3 +69,45 @@ func (cc *CacheClient) put(key string, commit *pfs.Commit) {
 	defer cc.mu.Unlock()
 	cc.cache.Add(key, commit)
 }
+
+func (cc *CacheClient) WithCreateFileSetClient(cb func(client.ModifyFile) error) (*pfs.CreateFileSetResponse, error) {
+	pachClient := pfsload.NewPachClient(cc.APIClient)
```
Comment (brycemcanally) [inline src/internal/pfssync/cache_client.go:74]: Why can't we just use the embedded APIClient?

File: src/server/worker/pipeline/transform/worker.go [hunk 2]
```diff
@@ -264,16 +264,16 @@ func handleDatumSet(driver driver.Driver, logger logs.TaggedLogger, task *DatumS
 	stats := &datum.Stats{ProcessStats: &pps.ProcessStats{}}
 	if err := pachClient.WithRenewer(func(ctx context.Context, renewer *renew.StringSet) error {
 		// Setup file operation client for output meta commit.
```
Comment (brycemcanally) [inline src/server/worker/pipeline/transform/worker.go:266]: very minor nit, could we move this comment down to the output meta commit `WithCreateFileSetClient` call.

File: src/internal/pfssync/cache_client.go (hunk 1 above)
Comment (FahadBSyed) [reply src/internal/pfssync/cache_client.go:74]: We can but I noticed, the `WithCreateFileSetClient()` function for `pfsload.Client` seems to wrap the `WithCreateFileSetClient()` for `APIClient` and adds a metadata() call to ensure that the metadata is configured properly. If that step is unnecessary then, we can instead use the embedded `APIClient` instead

Comment (brycemcanally) [reply src/internal/pfssync/cache_client.go:74]: That step is specific to the load testing tooling, the pach client available through the driver is already properly configured.

Comment (FahadBSyed) [reply src/internal/pfssync/cache_client.go:74]: Gotcha, so I will swap back to the `APIClient`

**Review by brycemcanally — APPROVED** [review]
lgtm

**Pull request merged**